FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m

```diff
@@ -62,6 +62,21 @@ - (void)populateNotificationContent:(UNMutableNotificationContent *)content
 }
 
 #if TARGET_OS_IOS || TARGET_OS_OSX || TARGET_OS_WATCH
+- (NSString *)fileExtensionForResponse:(NSURLResponse *)response {
+  NSString *const kDefaultExtension = @".tmp";
```

Comment:
I chose ".tmp" here because I wanted it to fail if there was no storage metadata included and no extension in the filename, but on reflection may make sense to default to ".png" which is allows us to at the very least make a simple assumption on the user's behalf rather than simply fail. Thoughts?